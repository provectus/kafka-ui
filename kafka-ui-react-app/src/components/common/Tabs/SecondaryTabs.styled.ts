import styled from 'styled-components';

export const SecondaryTabs = styled.nav`
  & button {
    background-color: ${(props) =>
      props.theme.secondaryTab.backgroundColor.normal};
    color: ${(props) => props.theme.secondaryTab.color.normal};
    padding: 6px;
    height: 32px;
    min-width: 57px;
    border: 1px solid ${(props) => props.theme.layout.stuffBorderColor};
    cursor: pointer;

    &:hover {
      background-color: ${(props) =>
        props.theme.secondaryTab.backgroundColor.hover};
      color: ${(props) => props.theme.secondaryTab.color.hover};
    }

    &.is-active {
      background-color: ${(props) =>
        props.theme.secondaryTab.backgroundColor.active};
      color: ${(props) => props.theme.secondaryTab.color.active};
    }
  }

  & > * {
    &:first-child {
      border-radius: 4px 0 0 4px;
    }
    &:last-child {
      border-radius: 0 4px 4px 0;
    }
    &:not(:last-child) {
      border-right: 0px;
    }
  }
`;
