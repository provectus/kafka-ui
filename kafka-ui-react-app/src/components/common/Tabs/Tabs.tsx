/* eslint-disable jsx-a11y/anchor-is-valid */
import React, { PropsWithChildren } from 'react';
import classNames from 'classnames';

interface TabsProps {
  tabs: string[];
  defaultSelectedIndex?: number;
  onChange?(index: number): void;
}

const Tabs: React.FC<PropsWithChildren<TabsProps>> = ({
  tabs,
  defaultSelectedIndex = 0,
  onChange,
  children,
}) => {
  const [selectedIndex, setSelectedIndex] =
    React.useState(defaultSelectedIndex);

  React.useEffect(() => {
    setSelectedIndex(defaultSelectedIndex);
  }, [defaultSelectedIndex]);

  const handleChange = (index: number) => {
    setSelectedIndex(index);
    onChange?.(index);
  };

  return (
    <>
      <div className="tabs">
        <ul>
          {tabs.map((tab, index) => (
            <li
              key={tab}
              className={classNames({ 'is-active': index === selectedIndex })}
            >
              <a
                role="button"
                tabIndex={index}
                onClick={() => handleChange(index)}
                onKeyDown={() => handleChange(index)}
              >
                {tab}
              </a>
            </li>
          ))}
        </ul>
      </div>
      {React.Children.toArray(children)[selectedIndex]}
    </>
  );
};

export default Tabs;
