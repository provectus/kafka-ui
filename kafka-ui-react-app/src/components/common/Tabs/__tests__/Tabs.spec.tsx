import React from 'react';
import Tabs from 'components/common/Tabs/Tabs';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

describe('Tabs component', () => {
  const tabs: string[] = ['Tab 1', 'Tab 2', 'Tab 3'];

  const child1 = <div className="child_1" />;
  const child2 = <div className="child_2" />;
  const child3 = <div className="child_3" />;

  render(
    <Tabs tabs={tabs}>
      {child1}
      {child2}
      {child3}
    </Tabs>
  );

  it('renders the tabs with default index 0', () => {
    expect(screen.queryAllByRole('listitem')[0]).toHaveClass('is-active');
  });
  it('renders the list of tabs', () => {
    screen.queryAllByRole('button').forEach((link, idx) => {
      expect(link).toHaveTextContent(tabs[idx]);
    });
  });
  it('renders the children', () => {
    screen.queryAllByRole('button').forEach((link, idx) => {
      userEvent.click(link);
      expect(screen.getByTestId(`.child_${idx + 1}`)).toBeInTheDocument();
    });
  });
  it('matches the snapshot', () => {
    const { container } = render(
      <Tabs tabs={tabs}>
        {child1}
        {child2}
        {child3}
      </Tabs>
    );
    expect(container).toBeInTheDocument();
  });
});
