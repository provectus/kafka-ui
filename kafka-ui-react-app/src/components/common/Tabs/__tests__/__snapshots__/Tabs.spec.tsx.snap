// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs component matches the snapshot 1`] = `
<Fragment>
  <div
    className="tabs"
  >
    <ul>
      <li
        className="is-active"
        key="Tab 1"
      >
        <a
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={0}
        >
          Tab 1
        </a>
      </li>
      <li
        className=""
        key="Tab 2"
      >
        <a
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={1}
        >
          Tab 2
        </a>
      </li>
      <li
        className=""
        key="Tab 3"
      >
        <a
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={2}
        >
          Tab 3
        </a>
      </li>
    </ul>
  </div>
  <div
    className="child_1"
    key=".0"
  />
</Fragment>
`;
